libbeat/cmd/instance/metrics.go

```diff
@@ -55,13 +59,29 @@ func setupMetrics(name string) error {
 		IncludeTop:   process.IncludeTopConfig{},
 	}
 	err := beatProcessStats.Init()
```

Comment:
Shouldn't all the initialization be done prior to registering the monitoring functions? Could there be a callback prior to setup completing.